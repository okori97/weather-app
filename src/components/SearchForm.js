import React from "react";
import "../styles/SearchForm.css";

export default function SearchForm({
	onSubmit,
	setSearchText,
	searchText,
	isWarm,
}) {
	const handleChange = (event) => {
		let input = event.target.value;
		setSearchText(input);
	};
	return (
		<form className="form">
			<input
				id="form__input"
				className="form__input"
				placeholder="Search a city..."
				type="search"
				onChange={handleChange}
			></input>
			<input
				className={"form__button " + (isWarm ? "warm " : "cold")}
				type="button"
				value={"Submit"}
				onClick={() => onSubmit()}
			></input>
		</form>
	);
}
