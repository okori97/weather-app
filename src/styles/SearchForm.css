.form__input {
  width: 100%;
  padding: 8px 8px 8px 8px;
  border: none;
  background: transparent;
}

.form__input:focus-visible {
  border: none;
}

.form__button {
  color: aliceblue;
  background-color: rgb(0, 0, 0);
  border: none;
  padding: 0px 16px 0px 16px;
  font-size: 14px;
  font-weight: 600;
}

.form {
  display: flex;
  margin-top: 8px;
  margin-bottom: 8px;
  border: 1px solid rgba(147, 145, 153, 0.356);
  /* border-radius: 80px; */
  max-width: 300px;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  transition: box-shadow 1s;
}

.form:hover {
  box-shadow:
    rgba(240, 46, 170, 0.4) -5px 5px,
    rgba(240, 46, 170, 0.3) -10px 10px,
    rgba(240, 46, 170, 0.2) -15px 15px,
    rgba(240, 46, 170, 0.1) -20px 20px,
    rgba(240, 46, 170, 0.05) -25px 25px;
}
