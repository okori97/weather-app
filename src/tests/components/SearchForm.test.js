import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Search Form", () => {
  const validProps = {
    setLocation: () => {},
    setForecasts: () => {},
    setSelectedDate: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        setLocation={validProps.setLocation}
        setForecasts={validProps.setForecasts}
        setSelectedDate={validProps.setSelectedDate}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  
});
