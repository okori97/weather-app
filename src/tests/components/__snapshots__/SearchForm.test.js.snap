// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders correctly 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <input
      class="form__input"
      id="form__input"
      placeholder="Search a city..."
      type="search"
    />
    <input
      class="form__button"
      type="button"
      value="Submit"
    />
  </form>
</DocumentFragment>
`;
